### PR TITLE
Create movie-search component, add to service, edit layout

### PR DESCRIPTION
**Pull request by evvis**
Implementation of movie search #6

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <app-header></app-header>
-<app-movie-list></app-movie-list>
+<router-outlet></router-outlet>
```
Comment (maximprokopchuk) [inline src/app/app.component.html]: Adding of routing is not related to this task, we will add it with separate PR later

File: src/app/app.module.ts [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
 import { HeaderComponent } from './header/header.component';
 import { MovieListComponent } from './movie-list/movie-list.component';
 import { MovieService } from './services/movie.service';
 import { MovieCardComponent } from './movie-list/movie-card/movie-card.component';
+import { MovieSearchComponent } from './movie-search/movie-search.component';
+
 
+const appRoutes: Routes = [
+  { path: '', redirectTo: '/home', pathMatch: 'full' },
+  { path: 'search/:searchValue', component: MovieSearchComponent },
```
Comment (maximprokopchuk) [inline src/app/app.module.ts]: This is wrong, search componenent should be used just in templates, it is simple view component

File: src/app/movie-list/movie-card/movie-card.component.html [hunk 3]
```diff
@@ -3,19 +3,21 @@
     <div class="card">
       <div class="card-image">
         <img *ngIf="movie.poster_path" src="https://image.tmdb.org/t/p/w500{{movie.poster_path}}">
-        <a class="btn-floating halfway-fab waves-effect waves-light red"><i class="material-icons">star</i></a>
         <div *ngIf="!movie.poster_path" >Load image</div>
       </div>
+      <div class="favorite-btn">
```
Comment (maximprokopchuk) [inline src/app/movie-list/movie-card/movie-card.component.html:8]: All the changes in this file are not related to the task. You can make any changes by separate PRs, not as part of this. Be more accurate

File: src/app/movie-list/movie-list.component.ts [hunk 4]
```diff
@@ -21,6 +21,19 @@ export class MovieListComponent implements OnInit {
...
+        });
+  }
+
+  searchMovie(searchValue: string){
+    this.movieService.searchMovie(searchValue)
+      .subscribe(
+        response => {
+          this.movieList = response;
+        });
+    console.log(searchValue);
```
Comment (maximprokopchuk) [inline src/app/movie-list/movie-list.component.ts]: Remove debug stuff

File: src/app/services/movie.service.ts [hunk 5]
```diff
@@ -5,21 +5,30 @@ import 'rxjs/add/operator/map';
...
+  public searchMovie(searchValue: string) {
+    let searchUrl = `${this.url}search/movie?api_key=${this.apiKey}&language=en-US&query=${searchValue}&page=1`;
+
+    return this.http.get(searchUrl)
+      .map(this.extractData);
+  }
+
   private extractData(res: Response) {
     let body = res.json();
+      console.log(res.json());
```
Comment (maximprokopchuk) [inline src/app/services/movie.service.ts]: remove debug stuff

File: src/app/services/search.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+interface Search {
```
Comment (maximprokopchuk) [inline src/app/services/search.ts]: This interface is wrong. In each query we get the same object shape than in get request, You could use Movie interface everywhere now. By the way this interface is not used at all, also Movie interface is defined but never used too. Fix that

**Review by maximprokopchuk — CHANGES_REQUESTED** [review]
You changed usage of Observables to usage of raw data in compoents, also you removed 'async' pipe in the template. This is not related change to in-progress task. We do any changes step by step. Any PR is introduce only one feature or fix only one bug, not many different changes in  one common PR. For now ok, let's do this, leave all changes Observable<data> -> data, but fix typing and see all my other comments

File: src/app/app.module.ts (hunk 2 above)
Comment (evvis) [reply src/app/app.module.ts]: This code remained after the experiments. Removed.

File: src/app/app.component.html (hunk 1 above)
Comment (evvis) [reply src/app/app.component.html]: This code remained after the experiments. Removed.

File: src/app/app.component.ts [hunk 7]
```diff
@@ -6,5 +6,5 @@ import { Component } from '@angular/core';
   styleUrls: ['./app.component.css']
 })
 export class AppComponent {
-  title = 'app';
+  title = 'TheMovieDB';
```
Comment (maximprokopchuk) [inline src/app/app.component.ts]: This var is not used. Remove it

**Review by maximprokopchuk — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts (hunk 7 above)
Comment (evvis) [reply src/app/app.component.ts]: Ok, remove

File: src/app/movie-list/movie-card/movie-card.component.html (hunk 3 above)
Comment (evvis) [reply src/app/movie-list/movie-card/movie-card.component.html:8]: this changes releated

File: src/app/movie-list/movie-list.component.ts (hunk 4 above)
Comment (evvis) [reply src/app/movie-list/movie-list.component.ts]: removed

File: src/app/services/movie.service.ts (hunk 5 above)
Comment (evvis) [reply src/app/services/movie.service.ts]: removed

File: src/app/services/search.ts (hunk 6 above)
Comment (evvis) [reply src/app/services/search.ts]: This interface remained after the experiments. Removed

**Comment by evvis** [comment]
Thank you for your comment. In the future I will stick. 
All chenges is done.

**Review by maximprokopchuk — APPROVED** [review]
Ok for now, but don't see any usage of Movie interface. It is not related change so will do it in separate PR
